### PR TITLE
Add a note in migration guide about the custom scalar compat mode

### PR DESCRIPTION
**Pull request by BoD**

### CODE REVIEW

**Review by martinbonnin — APPROVED** [review]
LGTM. I was thinking that we could maybe split the tutorial in 2?

1. Minimal changes to get things working again
2. Replacing all deprecated calls

**Comment by BoD** [comment]
Actually this is a good idea! Let me try and propose something.

File: docs/source/index.md [hunk 1]
```diff
@@ -19,6 +22,7 @@ Because generated types are query-specific, you can only access data that you ac
 * HTTP cache
 * File uploads
 * Custom scalar types
+* Supports Multiplatform projects
```
Comment (martinbonnin) [inline docs/source/index.md:25]: Move to 2nd position?

File: docs/source/get-started.mdx [hunk 2]
```diff
@@ -63,7 +63,7 @@ Now you can execute your defined query with `ApolloClient` like so:
 // or any other coroutine scope builder
 runBlocking {
   val response = try {
-    apolloClient.query(LaunchDetailsQuery(id = "83"))
+    apolloClient.query(LaunchDetailsQuery(id = "83")).execute()
```
Comment (martinbonnin) [inline docs/source/get-started.mdx:66]: 👍

File: docs/source/index.md [hunk 3]
```diff
@@ -4,9 +4,12 @@ description: A strongly-typed, caching GraphQL client for Java and Kotlin multip
 sidebar_title: Introduction
 ---
 
-[Apollo Android](https://github.com/apollographql/apollo-android) is a strongly-typed, caching GraphQL client for Android and Kotlin multiplatform apps, written in Kotlin. Despite its name, **you can use Apollo Android in any Java/Kotlin app.**
+[Apollo Android](https://github.com/apollographql/apollo-android) is a strongly-typed, caching GraphQL client for Android and Kotlin
+multiplatform apps, written in Kotlin. Despite its name, **you can use Apollo Android in any Java/Kotlin project.**
```
Comment (martinbonnin) [inline docs/source/index.md:8]: Did anything change here or am I crazy?

File: docs/source/index.md [hunk 4]
```diff
@@ -27,6 +31,7 @@ Apollo Android runs on the following platforms:
 * Android API level 15+
 * JDK 8+
 * iOS 13+
+* JavaScript
```
Comment (martinbonnin) [inline docs/source/index.md:34]: I changed this in `dev-3.x` already: https://github.com/apollographql/apollo-android/tree/dev-3.x#requirements

File: docs/source/migration/3.0.mdx [hunk 5]
```diff
@@ -16,7 +16,52 @@ Although most of the library's concepts are the same, many APIs have changed to
...
+1. Update your dependencies and imports (`com.apollographql.apollo` → `com.apollographql.apollo3`, see [section below](#package-name--group-id--plugin-id)).  
+  Remove `apollo-coroutines-support` and `apollo-android-support` if applicable.
+2. Gradle configuration:
+```kotlin
+apollo {
+  // Remove this
+  generateKotlinModels.set(true)
+  
+  // Add this
+  useVersion2Compat("com.example")
```
Comment (martinbonnin) [inline docs/source/migration/3.0.mdx:32]:
```suggestion
  useVersion2Compat()
```

The argument is `rootPackageName`, which most users will not have set  I believe (I know this is confusing, this is why we're making packageName mandatory in 3.x)

File: docs/source/migration/3.0.mdx [hunk 6]
```diff
@@ -16,7 +16,52 @@ Although most of the library's concepts are the same, many APIs have changed to
...
+  .query(MyQuery())
+  .await()
+
+// With:
+val response = apolloClient
+  .query(MyQuery())
+  .execute()
+```
+
+## All changes
```
Comment (martinbonnin) [inline docs/source/migration/3.0.mdx:62]:
```suggestion
## All details
```

File: docs/source/migration/3.0.mdx [hunk 7]
```diff
@@ -16,7 +16,52 @@ Although most of the library's concepts are the same, many APIs have changed to
 
 This page describes the most important changes, along with how to migrate an existing project from Apollo Android 2.x to Apollo Android 3.x.
 
-## Package name / group id / plugin id
+## The quick route 🚀
+
+Apollo Android 3.0 provides a few shortcuts and compatibility modes to ease the migration from 2.x.  To quickly arrive at a working state, follow these steps:
```
Comment (martinbonnin) [inline docs/source/migration/3.0.mdx:21]:
```suggestion
Apollo Android 3 provides a few helpers and compatibility modes to ease the migration from 2.x. To quickly reach a working state, follow the steps below.
Once you have a working app, we strongly advise to read the rest of this document as the helpers will be removed in a future version of Apollo Android.
```

Comment (martinbonnin) [reply docs/source/migration/3.0.mdx:21]: Also, we'll need to upgrade that text once we remove the helper is out to say something like "use 3.0 first and then migrate to 3.1"

File: docs/source/index.md (hunk 3 above)
Comment (martinbonnin) [reply docs/source/index.md]: Ahhh, that makes sense 👍 Sorry I missed this

**Pull request merged**